typed-racket-lib/typed-racket/base-env/base-env.rkt

```diff
@@ -2289,7 +2289,7 @@
 [special-comment-value (-> -Special-Comment Univ)]
 
 ;; Section 13.8
-[prop:custom-write -Struct-Type-Property]
+[prop:custom-write (-struct-property (-> -Self -Output-Port B ManyUniv))]
 [custom-write? (-> Univ B)]
 [custom-write-accessor (-> Univ (-> Univ -Output-Port B ManyUniv))]
```

Comment:
This would have to change as well, to use `(Un -True -False -Zero -One)` instead of `B` for the third argument of the result.